source/common/common/callback_impl.h

```diff
@@ -57,10 +57,8 @@ template <typename... CallbackArgs> class CallbackManager {
    * @param handle supplies the callback handle to remove.
    */
   void remove(CallbackHandle* handle) {
-    ASSERT(std::find_if(callbacks_.begin(), callbacks_.end(),
-                        [handle](const CallbackHolder& holder) -> bool {
-                          return handle == &holder;
-                        }) != callbacks_.end());
+    const auto comp = [handle](const CallbackHolder& holder) -> bool { return handle == &holder; };
+    ASSERT(std::find_if(callbacks_.begin(), callbacks_.end(), comp) != callbacks_.end());
```

Comment:
Re: moving lambda out of ASSERT, the TL;DR is don't ask, but if you must, see section 5.1.2 here http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4296.pdf

Reply:
reuse `comp` in the line below?

Reply:
interesting... I was actually trying to put long debug code in one assert with lambda like `ASSERT([]() { ...; return cond; } ());` so it is no longer possible.

btw, you can reuse `comp` in the line below.

Reply:
Could we have a version of ASSERT which would allow this to be done inline? It might be good to have that available for cases like this.